src/fsharp/NicePrint.fs

```diff
@@ -722,14 +722,18 @@ module private PrintTypes =
             let attrs = attrs |> List.filter (IsMatchingFSharpAttribute denv.g denv.g.attrib_ReflectedDefinitionAttribute >> not)
             let attrs = attrs |> List.filter (IsMatchingFSharpAttribute denv.g denv.g.attrib_StructLayoutAttribute >> not)
             let attrs = attrs |> List.filter (IsMatchingFSharpAttribute denv.g denv.g.attrib_AutoSerializableAttribute >> not)
+            
             match attrs with
             | [] -> restL 
             | _  -> squareAngleL (sepListL (rightL (tagPunctuation ";")) (List.map (layoutAttrib denv) attrs)) @@ 
                     restL
-        else 
-        match kind with 
-        | TyparKind.Type -> restL
-        | TyparKind.Measure -> squareAngleL (wordL (tagText "Measure")) @@ restL
+        elif Tastops.isStructRecordOrUnionTyconTy denv.g ty || 
+             ((Tastops.isUnionTy denv.g ty || Tastops.isRecdTy denv.g ty) && HasFSharpAttribute denv.g denv.g.attrib_StructAttribute attrs) then
```

Comment:
> return  false for [<Struct>] type DU = A

I need to understand this better.  It shouldn't return false, otherwise I don't think much would work in the compiler for those type definitions. Maybe I'm wrong, but are you sure ``A`` wasn't a type already defined, hence this was being interpreted as a type abbreviation?